### PR TITLE
rst to myst migration

### LINKED ISSUE #155038: Convert to markdown: torch.compiler_dynamo_overview.rst, torch.compiler_fake_tensor.rst, torch.compiler_faq.rst, torch.compiler_fine_grain_apis.rst, torch.compiler_get_started.rst
Convert the following files to MyST markdown and ensure that the doc tests are green:

- [torch.compiler_dynamo_overview.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/torch.compiler_dynamo_overview.rst)
- [torch.compiler_fake_tensor.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/torch.compiler_fake_tensor.rst)
- [torch.compiler_faq.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/torch.compiler_faq.rst)
- [torch.compiler_fine_grain_apis.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/torch.compiler_fine_grain_apis.rst)
- [torch.compiler_get_started.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/torch.compiler_get_started.rst)

To convert an `.rst` file:

1) Assign the issue to yourself using the `/assigntome` comment.

2) **Important!** Use `git mv filename.rst filename.md` to convert the issue from `.rst` to `.md`. **DO NOT delete**
   the `.rst` and create the `.md` - we won't accept PRs with deleted `.rst` files. We want to preserve the git log of
   those files. Therefore, using `git mv` is very important!

3) Before submitting a PR, build the docs locally as described in the [README](https://github.com/pytorch/pytorch/blob/main/README.md#building-the-documentation).
   Preview localy with `make serve` to ensure that the HTML renders correctly.    

4) Run `lintrunner` from the root of the repository:

   ```
   pip install lintrunner
   lintrunner init
   lintrunner -m main
   ```

4) Submit a PR with your changes and ensure the PR passes the doc CI tests. Specifically it needs to pass:

   * pull / linux-docs / build-docs-python-false (pull_request)
   * linux-jammy-py3.9-gcc11 / test (docs_test, 1, 1, lf.linux.2xlarge)

   Doc tests start after the Docker build, which might take about 30 minutes. 
   The doc build itself should take about 25 minutes. When the build is done, 
   you can see the preview of your build by clicking:
   📄 **Preview Python docs built from this PR** in the `pytorch-bot` comment.

   Make sure your PR is tagged with `docathon-h1-2025`, `module: docs`, `topic: not user facing`,
   and `topic: docs` labels.

5) Compare the preview of the pages you converted with same pages it the main branch.
   The converted pages in preview must look the same as in the main branch.

6) Once the PR is accepted, one of the maintainers will merge it.

See [Quick MyST Conversion Reference](https://github.com/pytorch/pytorch/wiki/Docstring-Guidelines#restructuredtext-vs-markdown)
in our wiki or [MyST Parser documentation](https://myst-parser.readthedocs.io/en/v0.15.1/sphinx/use.html) for detailed syntax.

Here is an example [PR](https://github.com/pytorch/pytorch/pull/154438)

cc @sekyondaMeta @AlannaBurke

### PR DESCRIPTION
**Pull request by Dhia-naouali**
Fixes #155038

(as I was able to build the html files from the myst md files I tried to host'em using github pages but I wasn't able to, still provided links to the html preview files in a different branch in case these would be useful)

converted files:
- [torch.compiler_dynamo_overview](https://github.com/pytorch/pytorch/blob/main/docs/source/torch.compiler_dynamo_overview.rst)
  - [original page](https://docs.pytorch.org/docs/stable/torch.compiler_dynamo_overview.html)
  - [my build](https://github.com/Dhia-naouali/pytorch/blob/docs-html-build-preview/preview/torch.compiler_dynamo_overview.html)

- [torch.compiler_fake_tensor](https://github.com/pytorch/pytorch/blob/main/docs/source/torch.compiler_fake_tensor.rst)
  - [original page](https://docs.pytorch.org/docs/stable/torch.compiler_fake_tensor.html)
  - [my html](https://github.com/Dhia-naouali/pytorch/blob/docs-html-build-preview/preview/torch.compiler_fake_tensor.html)

- [torch.compiler_faq](https://github.com/pytorch/pytorch/blob/main/docs/source/torch.compiler_faq.rst)
  - [original page](https://docs.pytorch.org/docs/stable/torch.compiler_faq.html)
  - [my html](https://github.com/Dhia-naouali/pytorch/blob/docs-html-build-preview/preview/torch.compiler_faq.html)

- [torch.compiler_fine_grain_apis](https://github.com/pytorch/pytorch/blob/main/docs/source/torch.compiler_fine_grain_apis.rst)
  - [original page](https://docs.pytorch.org/docs/stable/torch.compiler_fine_grain_apis.html)
  - [my html](https://github.com/Dhia-naouali/pytorch/blob/docs-html-build-preview/preview/torch.compiler_fine_grain_apis.html)

- [torch.compiler_get_started](https://github.com/pytorch/pytorch/blob/main/docs/source/torch.compiler_get_started.rst)
  - [original page](https://docs.pytorch.org/docs/stable/torch.compiler_get_started.html)
  - [my html](https://github.com/Dhia-naouali/pytorch/blob/docs-html-build-preview/preview/torch.compiler_get_started.html)

although through my own check the only difference is the date being mentioned in some of these pages (as a sub title), that wasn't created in the .rst file itself (most likely in by a configuration), I didn't add that as a frontmatter, lmk if I should do so

one last thing to mention is that the references didn't really work in my build (note: I didn't build the other files)

cc @svekars @sekyondaMeta @AlannaBurke

### CODE REVIEW

File: docs/source/torch.compiler_dynamo_overview.md [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+that provides performance comparison for different training backends. You can read
+more in the [TorchInductor post on PyTorch
+dev-discuss](https://dev-discuss.pytorch.org/t/torchinductor-a-pytorch-native-compiler-with-define-by-run-ir-and-symbolic-shapes/747).
+
+For an in-depth overview, read the sections below, watch the deep-dive video,
+and check out the dev-discuss topics.
+
+> - [Dynamo deep-dive video](https://www.youtube.com/watch?v=egZB5Uxki0I)
+> - [dev-discuss topics](https://dev-discuss.pytorch.org/search?q=TorchDynamo%20order%3Alatest)
+
```
Comment (svekars) [inline docs/source/torch.compiler_dynamo_overview.md]: Don't need the '>' sign
```suggestion
- [Dynamo deep-dive video](https://www.youtube.com/watch?v=egZB5Uxki0I)
- [dev-discuss topics](https://dev-discuss.pytorch.org/search?q=TorchDynamo%20order%3Alatest)
```

File: docs/source/torch.compiler_faq.md [hunk 2]
```diff
@@ -0,0 +1,637 @@
...
+`torch.compile` supports training, using AOTAutograd to capture backwards:
+
+1. The `.forward()` graph and `optimizer.step()` is captured by
+   TorchDynamo’s python `evalframe` frontend.
+2. For each segment of `.forward()` that torchdynamo captures, it uses
+   AOTAutograd to generate a backward graph segment.
+3. Each pair of forward and backward graph are (optionally) min-cut
+   partitioned to save the minimal state between forward and backward.
+4. The forward and backward pairs are wrapped in `autograd.function` modules.
+5. Usercode calling`.backward()` still triggers eager’s autograd engine,
```
Comment (svekars) [inline docs/source/torch.compiler_faq.md]:
```suggestion
5. User code calling `.backward()` still triggers eager’s autograd engine,
```

File: docs/source/torch.compiler_faq.md [hunk 3]
```diff
@@ -0,0 +1,637 @@
...
+are local fusions. So in practice this approach may be sufficient.
+
+## Do I still need to export whole graphs?
+
+For the vast majority of models you probably don’t and you can use
+`torch.compile()` as is but there are a few situations where
+full graphs are necessary and you can can ensure a full graph by simply
+running `torch.compile(..., fullgraph=True)`. These situations include:
+
+- Large scale training runs, such as \$250K+ that require pipeline parallelism
```
Comment (svekars) [inline docs/source/torch.compiler_faq.md]:
```suggestion
- Large scale training runs, such as $250K+ that require pipeline parallelism
```

**Review by svekars — COMMENTED** [review]
Thanks you so much! I added a couple of minor nits. But looking good overall - let's wait for preview to verify.

**Comment by Dhia-naouali** [comment]
> Thanks you so much! I added a couple of minor nits. But looking good overall - let's wait for preview to verify.

thanks for the thorough check @svekars ! I'll triple check it for anything that may have slipped through

File: docs/source/torch.compiler_faq.md [hunk 4]
```diff
@@ -0,0 +1,637 @@
...
+- **Excessive Recompilation**
+  When TorchDynamo compiles a function (or part of one), it makes certain
+  assumptions about locals and globals in order to allow compiler
+  optimizations, and expresses these assumptions as guards that check
+  particular values at runtime. If any of these guards fail, Dynamo will
+  recompile that function (or part) up to
+  `torch._dynamo.config.recompile_limit` times. If your program is
+  hitting the cache limit, you will first need to determine which guard is
+  failing and what part of your program is triggering it. The
+  [recompilation profiler](#recompilation-profiler) automates the
```
Comment (svekars) [inline docs/source/torch.compiler_faq.md:121]: #recompilation-profiler brakes the build and it looks like it's [not needed](https://github.com/pytorch/pytorch/issues/141422), so we can say something:

```suggestion
Use `TORCH_TRACE/tlparse` or `TORCH_LOGS=recompiles` to trace the root of the issue.
```
@williamwen42

File: docs/source/torch.compiler_faq.md [hunk 5]
```diff
@@ -0,0 +1,637 @@
...
+  process of setting TorchDynamo’s cache limit to 1 and running your
+  program under an observation-only ‘compiler’ that records the causes of
+  any guard failures. You should be sure to run your program for at least
+  as long (as many iterations) as you were running when you ran into
+  trouble, and the profiler will accumulate statistics over this duration.
+
```
Comment (svekars) [inline docs/source/torch.compiler_faq.md:127]:
```suggestion

```

File: docs/source/torch.compiler_faq.md [hunk 6]
```diff
@@ -0,0 +1,637 @@
...
+Triton in addition offers speedups because of automatic memory
+coalescing, memory management and scheduling within each Streaming
+Multiprocessor and has been designed to handle tiled computations.
+
+However, regardless of the backend you use it’s best to use a benchmark
+and see approach so try out the PyTorch profiler, visually inspect the
+generated kernels and try to see what’s going on for yourself.
+
+## Why am I not seeing speedups?
```
Comment (svekars) [inline docs/source/torch.compiler_faq.md:172]:
```suggestion
(torch.compiler_graph_breaks)=

## Why am I not seeing speedups?
```

File: docs/source/torch.compiler_faq.md [hunk 7]
```diff
@@ -0,0 +1,637 @@
...
+
+(torch-compiler-graph-breaks)=
```
Comment (svekars) [inline docs/source/torch.compiler_faq.md:174]:
```suggestion
```

File: docs/source/torch.compiler_faq.md (hunk 4 above)
Comment (williamwen42) [reply docs/source/torch.compiler_faq.md:121]: we can also link to the troubleshooting doc

**Comment by AlannaBurke** [comment]
There is a conflict that needs to be resolved.

**Comment by Dhia-naouali** [comment]
> There is a conflict that needs to be resolved.

this PR was actually split into two PRs (for sanity check): ([PR1/2](https://github.com/pytorch/pytorch/pull/155840) \& [PR2/2](https://github.com/pytorch/pytorch/pull/155911))
PR1/2 (which contain the [file causing the conflict](https://github.com/pytorch/pytorch/blob/main/docs/source/torch.compiler_dynamo_overview.md)) was merged (/labeled as merged) few hours ago), which is causing the conflict with this one 
I thought I would have to close this parent PR once the other two get merged/ closed (or should I close it rn ?)
